src/main/java/edu/umich/ctools/sectionsUtilityTool/Utils.java

```diff
@@ -182,4 +193,56 @@ private static void logMsg(String uniqname, String originalUrl, HttpServletReque
 		M_log.info(String.format(baseString, user, originalUrl));
 	}
 
+	public static String makeApiCall(String... urlChunks){
+		String url = canvasURL + CANVAS_API_VERSION;
+		for (String chunk : urlChunks) {
```

Comment:
Would "url += chunk;" work?  I think that avoids reallocating strings with each join.